src/hooks/useFeedbackMessage/useFeedbackMessage.tsx

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright 2023 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useCallback, useMemo } from 'react'
+import { message } from 'antd'
+
+import { FeedbackMessage } from './useFeedbackMessage.types'
+import { FeedbackMessageProps } from './useFeedbackMessage.props'
+import { Message } from '../../components/Message'
+
+type FeedbackMessageType = 'loading' | 'info' | 'success' | 'warning' | 'error'
+
+/**
+ * A hook that allow to display global informative messages to the user at the top of the page. It is intended
+ * to give user feedbacks on a certain operation executed or information regarding the context of the application.
+ *
+ * The hook returns several methods to be used to show a message of different type
+ * (each type will use a different icon) or to manually remove rendered messages.
+ *
+ * @example
+ * export const Component(props: {...}) => {
+ *  // When called, show a FeedbackMessagea
+ *  const onClick = (): void => {
+ *    success({
+ *      extra: <Button size={Size.Small} onClick={onDismiss}>Close</Button>,
+ *      key: 'messageKey',
+ *      message: 'This is a feedback message',
+ *    })
+ *  }
+ *
+ *  // When called, remove the FeedbackMessage, if still there
+ *  const onDismiss = (): void => { dismiss('messageKey') }
+ *
+ *  return (
+*      <div>
+*       {...}
+*       <Button onClick={onClick}>Click me to show a Feedback Message</Button>
+*      </div>
+ *  )
+ * }
+ *
+ * @returns {FeedbackMessage} An object including several functions to call to
+ * manage the rendering of feedback messages.
+ */
```

Comment:
```suggestion
/**
 * A hook that allows to display global informative messages to the user at the top of the page. It is intended to give
 * feedback messages on a certain operation being executed or information regarding the context of the application.
 *
 * The hook returns several methods to be used to show a message of a different type
 * (each type will use a different icon) or to manually remove rendered messages.
 *
 * @example
 * export const Component(props: {...}) => {
 *   // When called, shows a FeedbackMessage
 *   const onClick = (): void => {
 *     success({
 *       extra: <Button size={Size.Small} onClick={onDismiss}>Close</Button>,
 *       key: 'messageKey',
 *       message: 'This is a Feedback Message',
 *     })
 *   }
 *
 *   // When called, removes the FeedbackMessage, if still visible
 *   const onDismiss = (): void => { dismiss('messageKey') }
 *
 *   return (
 *     <div>
 *       {...}
 *       <Button onClick={onClick}>Click me to show a Feedback Message</Button>
 *     </div>
 *   )
 * }
 *
 * @returns {FeedbackMessage} An object which includes several functions to
 * manage the rendering of feedback messages.
 */
```

Reply:
Is the example really necessary, anyway?

Reply:
I like the idea of giving an user a gist on how to use the component (without relying on an external link - as storybook) for quickly check how to use the hook within your editor.

Not complaining if you ask me to remove it tho.